File: nom-api/Nom.Orch/Models/Pantry/ShoppingNeedsResponseModel.cs
using System;
using System.Collections.Generic;

namespace Nom.Orch.Models.Pantry
{
    /// <summary>
    /// Response for the shopping needs endpoint.
    /// </summary>
    public class ShoppingNeedsResponseModel
    {
        public long HouseholdId { get; set; }
        public int DaysAhead { get; set; }
        public DateOnly FromDate { get; set; }
        public DateOnly ToDate { get; set; }
        public int MealCount { get; set; }
        public List<ShoppingNeedModel> Needs { get; set; } = new();
    }
}

File: nom-ui/src/app/core/models/shopping-needs-response.model.ts
import { ShoppingNeedItem } from './shopping-need-item.model';

export interface ShoppingNeedsResponse {
  householdId: number;
  daysAhead: number;
  fromDate: string;
  toDate: string;
  mealCount: number;
  needs: ShoppingNeedItem[];
}
